src/components/workspace/ReplControl.tsx

```diff
@@ -14,48 +13,47 @@ export interface IReplControlProps {
   handleChapterSelect: (e: React.ChangeEvent<HTMLSelectElement>) => void
 }
 
-const chapterSelect = (handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {}) => (
-  <div className="pt-select pt-select">
-    <select defaultValue={sourceChapters.slice(-1)[0].toString()} onChange={handleSelect}>
-      {sourceChapters.map(chap => (
-        <option key={chap} value={chap}>
-          {`Source \xa7${chap}`}
-        </option>
-      ))}
-    </select>
-  </div>
-)
-
 class ReplControl extends React.Component<IReplControlProps, {}> {
   public render() {
-    const genericButton = (
-      label: string,
-      icon: IconName,
-      handleClick = () => {},
-      intent = Intent.NONE,
-      notMinimal = false
-    ) => (
-      <Button
-        onClick={handleClick}
-        className={(notMinimal ? '' : 'pt-minimal') + ' col-xs-12'}
-        intent={intent}
-        icon={icon}
-      >
-        {label}
-      </Button>
-    )
-    const evalButton = genericButton('', 'code', this.props.handleReplEval)
-    const clearButton = genericButton('', 'remove', this.props.handleReplOutputClear)
     return (
       <div className="row end-xs">
         <div className="pt-control-group pt-fill">
           {chapterSelect(this.props.handleChapterSelect)}
-          {evalButton}
-          {clearButton}
+          {controlButton('', 'code', this.props.handleReplEval)}
+          {controlButton('', 'remove', this.props.handleReplOutputClear)}
         </div>
       </div>
     )
   }
 }
 
+const controlButton = (
```

Comment:
Nice, can this be applied to `Editor` as well? I think it still uses `genericButton`

Reply:
Sounds good (and DRY), but how do you wanna go about it?

1. Change `Editor` to have the same function `controlButton`
2. .Export `controlButton` from here and import it into `Editor`
3. Abstract `controlButton` into a separate file within `workspace`
4. Abstract `controlButton` into a new/existing file under `utils`

I'm leaning towards (2) tbh

Reply:
Since `controlButton` does not originate from here, I think abstracting into the `utils` folder is better. If there was something that belongs to `ReplControl`, it would make more sense to export.